enhancements/cluster-logging/fine-grained-per-namespace-logs-access.md

```diff
@@ -0,0 +1,249 @@
+---
+title: fine-grained-logs-access
+
+authors:
+  - "@aminesnow"
+
+reviewers:
+  - "@periklis"
+  - "@shwetaap"
+  - "@alanconway"
+  - "@xperimental"
+  - "@jcantrill"
+
+approvers:
+  - "@periklis"
+
+creation-date: 2023-05-31
+last-updated: 2023-05-31
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/LOG-4020
+see-also: []
+replaces: []
+superseded-by: []
+---
+
+# Openshift Logging: Fine grained access to logs
+
+## Summary
+
+This enhancement proposal seeks to outline the solution for a more fine grained access to LokiStack logs on OpenShift Container Platform 4, ie, have the ability as the cluster administrator to grant access to logs on a namespace basis.
+
+The proposal provides an overview of the changes required to achieve this goal as well as some implementation details.
+
+## Motivation
+
+In enterprise environments, where OpenShift is used across different legal entities, it's common to have central teams that support the application teams in the respective entities. But given that some application may log sensitive data, those centralized support teams are not granted access to logs but they can only view specific objects, such as pods in the namespace.
+
+Currently, access to logs in LokiStack is granted when a user has access to the given namespace or when the user is part of a specific cluster-admin Group. And even though OpenShift Container Platform 4 does allow to configure RBAC to address this issue, LokiStack does not, and therefore grants access to logs to people that should not see them.
+
+This enhancement proposal presents a solution that enables cluster admins to have a more fine grained control over who accesses what logs.
+
+### User Stories
+
+* Use Case number one, is where a user has access to a specific namespace to see the objects included but is denied to view logs from pods (for legal reason mostly).
+
+* Use Case number two, is where a number of people have elevated permissions and therefore are able to access pretty much all namespaces on the OpenShift cluster. Usually those people are responsible for enabling and supporting applications when onboarding to OpenShift and therefore have permissions to see most of the objects in the given namespaces. However, due to legal and data protection constrains, those users can not have access to logs and therefore LokiStack should prevent them from seeing application specific logs.
+
+### Goals
+
+* Cluster admins can grant and revoke access of the workload logs of specific namespaces to users.
```

Comment:
Not sure if its additional goal or rewording but fundamentally we probably desire to make use of OpenShift RBAC in order that we are consistent with how permissions are managed elsewhere in lieu of  the contrived way we utilize now (e.g. Elasticsearch perms)